server/polar/repository/service.py

```diff
@@ -180,6 +180,28 @@ async def update_badge_settings(
 
         return settings
 
+    async def update_settings(
+        self,
+        session: AsyncSession,
+        repository: Repository,
+        settings: RepositoryUpdateProfile,
+    ) -> Repository:
+        if settings.profile_settings is not None:
+            repository.profile_settings = {
+                **repository.profile_settings,
+                **settings.profile_settings.model_dump(mode="json", exclude_unset=True),
+            }
+
+        updated = await repository.save(session)
```

Comment:
I'm currently on a effort to remove the `.save` pattern in #2805. I would recommend instead:

```py
session.add(repository)
await session.flush()
```

Reply:
Nice! Fixed :)